src/core/turn-order.test.js

```diff
@@ -144,3 +147,29 @@ test('override', () => {
   state = flow.processGameEvent(state, { type: 'endTurn' });
   expect(state.ctx.currentPlayer).toBe('5');
 });
+
+test('custom order', () => {
+  const flow = FlowWithPhases({
+    phases: [
+      {
+        name: 'A',
+        turnOrder: TurnOrder.DEFAULT,
+        onPhaseBegin: (G, ctx) => (ctx.playerOrder = [2, 0, 1]),
```

Comment:
Can you set `state.ctx.playerOrder` down in line 167 (instead of passing `undefined`) and remove `onPhaseBegin`?

Reply:
I've tried but wasn't able to move this. What if we wait for events to manage this test properly?

Reply:
I fixed this for you.